### PR TITLE
Add aten::repeat_interleave operator with layer test

### PR DESCRIPTION
**Pull request by sikorsl1**
**Details:**

- `aten::repeat_interleave` operator has been added
- layer test for this operator has been added

**Ticket:**

- 99228

### CODE REVIEW

File: src/frontends/pytorch/src/op/repeat_interleave.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset8.hpp"
```
Comment (mmikolajcz) [inline src/frontends/pytorch/src/op/repeat_interleave.cpp]: There was a switch from opset8 to opset10

File: src/frontends/pytorch/src/op/repeat_interleave.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            // case (repeats=tensor, dim=None)
+            auto flat_shape = context.mark_node(opset8::Constant::create(element::i64, Shape{1}, {-1}));
+            auto reshape = std::make_shared<opset8::Reshape>(input, flat_shape, false);
+            OutputVector all_indices = generate_indices_from_repeats_tensor(repeats, context);
+            auto concat = std::make_shared<opset8::Concat>(all_indices, 0);
+            result = std::make_shared<opset8::Gather>(reshape, concat, const_0);
+        }
+    } else {
+        auto dim = context.const_input<int64_t>(2);
+        auto const_dim = context.mark_node(opset8::Constant::create(element::i64, Shape{}, {dim}));
```
Comment (mmikolajcz) [inline src/frontends/pytorch/src/op/repeat_interleave.cpp]: Wouldn't it be better to load dim directly from input (`context.get_input(2)`) instead loading it into cpp variable and then creating OV const?

**Review by mmikolajcz — CHANGES_REQUESTED** [review]

File: src/frontends/pytorch/src/op/repeat_interleave.cpp (hunk 1 above)
Comment (sikorsl1) [reply src/frontends/pytorch/src/op/repeat_interleave.cpp]: Thanks, done.

File: src/frontends/pytorch/src/op/repeat_interleave.cpp (hunk 2 above)
Comment (sikorsl1) [reply src/frontends/pytorch/src/op/repeat_interleave.cpp]: Of course, thanks for that comment :) Done.

**Review by mmikolajcz — APPROVED** [review]

**Review by bszmelcz — APPROVED** [review]
@sikorsl1 It is not necessary but I think that it is good idea to have 2 test functions. First one with random initialize arrays and the second one with hard coded ones with values taken from the documentation of a given operator. An example might be `openvino/tests/layer_tests/pytorch_tests/test_div.py.`

**Pull request closed without merging**